File: api/krameriusplus-core-v2/src/main/java/cz/inqool/dl4dh/krameriusplus/corev2/request/export/request/ExportRequest.java
package cz.inqool.dl4dh.krameriusplus.corev2.request.export.request;

import cz.inqool.dl4dh.krameriusplus.corev2.job.KrameriusJobInstance;
import cz.inqool.dl4dh.krameriusplus.corev2.job.config.export.ExportJobConfig;
import cz.inqool.dl4dh.krameriusplus.corev2.request.Request;
import cz.inqool.dl4dh.krameriusplus.corev2.request.export.item.ExportRequestItem;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
@Entity
@Table(name = "kplus_export_request")
public class ExportRequest extends Request {

    @OneToOne(cascade = CascadeType.ALL, orphanRemoval = true)
    private ExportJobConfig config;

    @OneToMany(mappedBy = "exportRequest")
    private List<ExportRequestItem> items = new ArrayList<>();

    @OneToOne
    @NotNull
    private KrameriusJobInstance createRequestJob;

    @Embedded
    @NotNull
    private BulkExport bulkExport;
}

File: web/src/models/request/export-request.ts
import { ExportJobConfig } from 'models/job/config/export-job-config'
import { KrameriusJobInstance } from 'models/job/kramerius-job-instance'
import { BulkExport } from '../bulk-export'
import { ExportRequestItem } from './export-request-item'
import { Request } from './request'

export interface ExportRequest extends Request {
	config: ExportJobConfig
	items: ExportRequestItem[]
	createRequestJob: KrameriusJobInstance
	bulkExport?: BulkExport
}
